controllers/activemqartemis_controller.go

```diff
@@ -526,6 +530,9 @@ func AssertConfigMapContainsKey(configMap corev1.ConfigMap, key string, contextM
 }
 
 func AssertSecretContainsKey(secret corev1.Secret, key string, contextMessage string) *metav1.Condition {
+	if certutil.IsSecretFromCert(&secret) {
+		return nil
```

Comment:
if a cert manager secret is not fully initialised, as in some signing is still in progress. do we want to address that here?

Reply:
cert manager will annotate as it makes progress, maybe we need to be aware of that workflow and only reconcile when the bits we need are there. just a thought!

Reply:
+1, will add a check on the annotation to check the progress.

Reply:
I didn't find a suitable annotation for that purpose on secret. I added a further check on the secret that it has to have the three keys in its data (i.e. ca.crt, tls.crt, and tls.key).